clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -182,14 +183,20 @@ public void done(long baseOffset, long logAppendTime, RuntimeException exception
         Iterator<MutableRecordBatch> recordBatchIter = memoryRecords.batches().iterator();
         if (!recordBatchIter.hasNext())
             throw new IllegalStateException("Cannot split an empty producer batch.");
+
         RecordBatch recordBatch = recordBatchIter.next();
+        if (recordBatch.magic() < MAGIC_VALUE_V2 && !recordBatch.isCompressed())
```

Comment:
@becketqin Actually, in my original patch, I modified this code to handle all cases. Perhaps it would be a little simpler to revert to that behavior?